src/tools/compiletest/src/runtest.rs

```diff
@@ -1334,8 +1334,18 @@ actual:\n\
         // FIXME (#9639): This needs to handle non-utf8 paths
         let mut args = vec![input_file.to_str().unwrap().to_owned(),
                             "-L".to_owned(),
-                            self.config.build_base.to_str().unwrap().to_owned(),
-                            format!("--target={}", target)];
+                            self.config.build_base.to_str().unwrap().to_owned()];
+
+        // Optionally prevent default --target if specified in test compile-flags.
+        let custom_target = self.props.compile_flags
+            .iter()
+            .fold(false, |acc, ref x| acc || x.starts_with("--target"));
+
+        if !custom_target {
+            args.extend(vec![
+                format!("{}", target),
```

Comment:
Sorry about that.
Definitely a bug. I was having some issues with the tests, sometimes all passing (possibly because I wasn't recompiling rustc correctly), so I didn't pick that up.

Hopefully the latest commit addresses this.